packages/react-native-oursky/src/VerifyOTP.js

```diff
@@ -0,0 +1,219 @@
+    shadowOffset: { width: 0, height: 10 },
+    shadowOpacity: 0.2,
+    elevation: 15,
+    paddingHorizontal: 21,
+    paddingVertical: 31,
+  },
+  description: {
+    fontSize: 18,
+    lineHeight: 24,
+    color: "rgb(51, 51, 51)",
+    marginBottom: 23,
+  },
+  codeBoxContainer: {
+    flexDirection: "row",
+    marginHorizontal: -4.5,
+  },
+  error: {
+    color: "rgb(238, 0, 0)",
+    alignSelf: "center",
+    marginTop: 12,
+  },
+  resendContainer: {
+    marginTop: 33,
+    alignItems: "center",
+  },
+  resendText: {
+    fontSize: 13,
+    color: "rgb(170, 170, 170)",
+  },
+  hiddenTextInput: {
+    width: 0,
+    height: 0,
+    opacity: 0,
+  },
+});
+
+export type Props = ExtraTextProps & {
+  description: React.Node,
+  resendText: React.Node,
+  resending: boolean, // disabled resend button
+  countDownFrom: number,
+
+  style?: ViewStyle,
+  descriptionStyle?: TextStyle,
+  codeBoxStyle?: ViewStyle,
+  codeBoxTextStyle?: TextStyle,
+  resendContainerStyle?: ViewStyle,
+  resendTextStyle?: TextStyle,
+
+  onEnterCode?: (code: string, clearCode: () => void) => void,
+  onPressResend?: (restartTimer: () => void) => void,
+};
+
+type State = {
+  value: string,
+  countDownSecond: number,
+};
+
+type TextInputRefProps = {
+  textInputRef: React$Ref<TextInput>,
+};
+
+class VerifyOTP extends React.PureComponent<Props & TextInputRefProps, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: "",
+      countDownSecond: props.countDownFrom,
+    };
+  }
+
+  timerId: IntervalID | null = null;
+
+  componentDidMount() {
+    this.countDown();
+  }
+
+  componentWillUnmount() {
+    if (this.timerId) {
+      clearInterval(this.timerId);
+    }
+  }
+
+  countDown = () => {
+    this.setState({ countDownSecond: this.props.countDownFrom });
+    if (this.timerId) {
+      clearInterval(this.timerId);
+    }
+    this.timerId = setInterval(this.countDownUntilZero, 1000);
+  };
+
+  countDownUntilZero = () => {
+    this.setState(
+      prevState => ({
+        countDownSecond: prevState.countDownSecond - 1,
+      }),
+      () => {
+        if (this.state.countDownSecond === 0 && this.timerId) {
+          clearInterval(this.timerId);
+          this.timerId = null;
+        }
+      }
+    );
+  };
+
+  clearCode = () => {
+    this.setState({ value: "" });
+  };
+
+  onChangeText = (value: string) => {
+    if (/^\d*$/.test(value)) {
+      this.setState({ value }, () => {
+        if (this.props.onEnterCode) {
+          this.props.onEnterCode(value, this.clearCode);
+        }
+      });
+    }
+  };
+
+  onPressResend = () => {
+    if (this.props.onPressResend) {
+      this.props.onPressResend(this.countDown);
+    }
+  };
+
+  renderCodeBox = () => {
+    const { codeBoxStyle, codeBoxTextStyle } = this.props;
+    return (
+      <View style={defaultStyles.codeBoxContainer}>
+        {Array(4)
+          .fill("")
+          .map((_, idx) => {
+            return (
+              <CodeBox
```

Comment:
@shana0440 TextInput should be focused when pressed either one of the code boxes